Lib/test/test_builtin.py

```diff
@@ -457,6 +457,33 @@ def test_divmod(self):
             self.assertAlmostEqual(result[1], exp_result[1])
 
         self.assertRaises(TypeError, divmod)
+        self.assertRaises(TypeError, divmod, 12, 7, 3)
+        self.assertRaises(TypeError, divmod, 12, [])
+        self.assertRaises(TypeError, divmod, [], 7)
+
+        def fake_divmod(divmod_ret_val):
+            class FakeDivmod:
+                def __divmod__(*args):
+                    return divmod_ret_val
+            return FakeDivmod()
+        def fake_rdivmod(divmod_ret_val):
+            class FakeRDivmod:
+                def __rdivmod__(*args):
+                    return divmod_ret_val
+            return FakeRDivmod()
+
+        self.assertEqual(divmod(fake_divmod((2, 3)), []), (2, 3))
+        with self.assertWarns(DeprecationWarning):
+            self.assertEqual(divmod(fake_divmod([2, 3]), []), (2, 3))
+        self.assertRaises(TypeError, divmod, fake_divmod(42), [])
+        self.assertRaises(TypeError, divmod, fake_divmod(()), [])
+        self.assertRaises(TypeError, divmod, fake_divmod((1, 2, 3)), [])
+        self.assertEqual(divmod([], fake_rdivmod((2, 3))), (2, 3))
+        with self.assertWarns(DeprecationWarning):
+            self.assertEqual(divmod([], fake_rdivmod([2, 3])), (2, 3))
+        self.assertRaises(TypeError, divmod, [], fake_rdivmod(42))
+        self.assertRaises(TypeError, divmod, [], fake_rdivmod(()))
+        self.assertRaises(TypeError, divmod, [], fake_rdivmod((1, 2, 3)))
```

Comment:
Might be good to add a `namedtuple` example here, to state explicitly that tuple subclasses are intended to be supported.

Reply:
I hesitate. Should we allow tuple subclasses or require exact tuples? Most converting special methods (`__int__`, `__index__`, `__float__`, `__complex__`) are required to return an instance of exact type. From C side, if use the C API for unpacking the tuple, there is no difference between tuples and tuple subclasses. But there mey be difference at Python level if the tuple subclass overrides some methods like `__add__`, `__iter__`, `__len__`, `__getitem__`.

For now, I prefer to not promise the namedtuple support.